api/query/query.go

```diff
@@ -97,17 +135,17 @@ func (mc memcacheReadWritable) NewReadCloser(key string) (io.ReadCloser, error)
 }
 
 func (mc memcacheReadWritable) NewWriteCloser(key string) (io.WriteCloser, error) {
```

Comment:
@mdittmer I know you've merged the PR but still want to share a thought on this:

The fact that changing the return value doesn't change the function signature reminds me of a Go convention that we didn't follow here: take *interfaces* as arguments, and return *concrete* types. In this example, we should return `*memcacheWriteCloser`. Explicitness in return values benefits the callers (and callers don't have to care - they can always use `:=` or even assign the return value to an interface type), while the polymorphism in arguments makes the function more generic.